From 3e7ae73a61a5dcdc55103ee8e983ab46beb063d0 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:31 -0400
Subject: [block] blk-mq: fix sparse warning on missed __percpu annotation

Message-id: <1402670238-13916-104-git-send-email-snitzer@redhat.com>
Patchwork-id: 83904
O-Subject: [RHEL7.1 PATCH 103/150] blk-mq: fix sparse warning on missed __percpu annotation
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit e6cdb0929fe6726ba5203fc5529b74564d98a9e9
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Tue Jun 3 11:24:06 2014 +0800

    blk-mq: fix sparse warning on missed __percpu annotation

    'struct blk_mq_ctx' is  __percpu, so add the annotation
    and fix the sparse warning reported from Fengguang:

     [block:for-linus 2/3] block/blk-mq.h:75:16: sparse: incorrect
     type in initializer (different address spaces)

    Reported-by: kbuild test robot <fengguang.wu@intel.com>
    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 43eb315..3bb4cfe 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1767,7 +1767,7 @@ static void blk_mq_add_queue_tag_set(struct blk_mq_tag_set *set,
 struct request_queue *blk_mq_init_queue(struct blk_mq_tag_set *set)
 {
  struct blk_mq_hw_ctx **hctxs;
- struct blk_mq_ctx *ctx;
+ struct blk_mq_ctx __percpu *ctx;
  struct request_queue *q;
  unsigned int *map;
  int i;
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 5b59b1c..d92da6e 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -351,7 +351,7 @@ struct request_queue {
  unsigned int  *mq_map;
 
  /* sw queues */
- struct blk_mq_ctx *queue_ctx;
+ struct blk_mq_ctx __percpu *queue_ctx;
  unsigned int  nr_queues;
 
  /* hw dispatch queues */
-- 
1.7.1